tests/components/mqtt/test_init.py

```diff
@@ -1336,6 +1369,8 @@ async def test_not_calling_unsubscribe_with_active_subscribers(
     unsub = await mqtt.async_subscribe(hass, "test/state", record_calls)
     await mqtt.async_subscribe(hass, "test/state", record_calls)
     await hass.async_block_till_done()
+    async_fire_time_changed(hass, utcnow() + timedelta(SUBSCRIBE_COOLDOWN))
```

Comment:
For this test applies the same, we are asserting on the MQTT client. We should not mock the HA MQTT client code.